### PR TITLE
add type checks for backticks_to_code_tags

### PR DESCRIPTION
**Pull request by yangchristina**

### CODE REVIEW

**Review by firasm — CHANGES_REQUESTED** [review]
This looks like a reasonable change, but can you add a [test](https://github.com/open-resources/problem_bank_helpers/blob/main/tests/test_problem_bank_helpers_other_functionality.py) for this particular case? 

It should fail with the old code, and pass with the new code.

**Comment by firasm** [comment]
You can request a review from @Bluesy1 for the test (or if you need a hand setting it up).

**Comment by Bluesy1** [comment]
Seems like a reasonable change (xref https://ca.prairielearn.com/pl/course/38/question/270667/preview?variant_id=12765502)
A simple case should be the following dictionary I think:
```python
case = {"params": {"part1": 1, "part2": {"ans1": {1}}}
```

**Comment by yangchristina** [comment]
@Bluesy1 are these tests okay? I haven't used pytest before, also I don't know how to run it.

File: tests/test_problem_bank_helpers_other_functionality.py [hunk 1]
```diff
@@ -64,4 +65,38 @@ def test_missing_values(file_path):
...
+                "statement": True,
+                "something": 2,
+            }
+        },
+    }
+
+    try:
+        pbh.backticks_to_code_tags(data)
+    except Exception as exc:
+        assert False, f"Raised an exception {exc}"
```
Comment (Bluesy1) [inline tests/test_problem_bank_helpers_other_functionality.py]:
```suggestion
    pbh.backticks_to_code_tags(data)

```

**Review by Bluesy1 — CHANGES_REQUESTED** [review]
No need to reraise the exception - an exception getting raised already causes the test to fail. (`assert False` will just raise an AssertionError instead of whatever Exception was otherwise raised, while potentially quashing context)

Otherwise, looks good to me!

**Review by Bluesy1 — APPROVED** [review]
Thanks for the quick update!

I appreciate that pytest can be a bit difficult to wrap your head around at first, nice job!

**Comment by yangchristina** [comment]
@Bluesy1 great, thanks for your help!

**Review by firasm — APPROVED** [review]

**Pull request merged**